### PR TITLE
UI, documentation, and functionality improvements for email templates merge to core

### PR DESCRIPTION
**Pull request by kimcoleman**
### All Submissions:

* [x] Have you followed the [Contributing guideline](https://github.com/strangerstudios/paid-memberships-pro/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/strangerstudios/paid-memberships-pro/pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Improvements to layout of the Memberships > Settings > Email Templates admin page.
* Fix styling of the template updates, errors, etc. messages shown on the form processing.
* Fixed typo in the subject line for word "udpated" => "updated". Note: the language packs need to be rebuilt for this, I only modified the default.
* Added fallback option to get_user_by email address before fallback to current_user for the `pmpro_email_templates_email_data` function.
* Updated the message on deprecated Add Ons to be deleted.
* Added a few template variable replacements to emails where that data is and should be available.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Review by andrewlimaza — APPROVED** [review]

File: adminpages/emailtemplates.php [hunk 1]
```diff
@@ -19,229 +19,244 @@
...
+					</div>
+				</td>
+			</tr>
+			<tr>
+				<th scope="row" valign="top">
+					<label for="pmpro_email_template_switcher"><?php esc_html_e( 'Email Template', 'paid-memberships-pro' ); ?></label>
+				</th>
+				<td>
+					<select name="pmpro_email_template_switcher" id="pmpro_email_template_switcher">
+						<option value="" selected="selected"><?php echo '--- ' . __( 'Select a Template to Edit', 'paid-memberships-pro' ) . ' ---'; ?></option>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:41]:
```suggestion
						<option value="" selected="selected"><?php echo '--- ' . esc_html__( 'Select a Template to Edit', 'paid-memberships-pro' ) . ' ---'; ?></option>

```

File: adminpages/emailtemplates.php [hunk 2]
```diff
@@ -19,229 +19,244 @@
...
+					<?php foreach ( $pmpro_email_templates_defaults as $key => $template ): ?>
+						<option value="<?php echo $key; ?>"><?php echo $template['description']; ?></option>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:43]:
```suggestion
						<option value="<?php echo esc_attr( $key ); ?>"><?php echo esc_html( $template['description'] ); ?></option>

```

File: adminpages/emailtemplates.php [hunk 3]
```diff
@@ -19,229 +19,244 @@
...
+					<?php endforeach; ?>
+					</select>
+					<img src="<?php echo admin_url( 'images/wpspin_light.gif' ); ?>" id="pmproet-spinner" style="display:none;"/>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:46]:
```suggestion
					<img src="<?php echo esc_url( admin_url( 'images/wpspin_light.gif' ) ); ?>" id="pmproet-spinner" style="display:none;"/>

```

File: adminpages/emailtemplates.php [hunk 4]
```diff
@@ -19,229 +19,244 @@
...
+					<p id="email_template_help_text" class="description"></p>
+				</td>
+			</tr>
+			<tr class="hide-while-loading">
+				<th scope="row" valign="top"></th>
+				<td>
+					<label><input id="email_template_disable" name="email_template_disable" type="checkbox"/><span
+							id="disable_label"><?php esc_html_e('Disable this email?', 'paid-memberships-pro');?></span></label>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:54]:
```suggestion
							id="disable_label"><?php esc_html_e('Disable this email?', 'paid-memberships-pro'); ?></span></label>

```

File: adminpages/emailtemplates.php [hunk 5]
```diff
@@ -19,229 +19,244 @@
...
+
+					<p id="disable_description" class="description"><?php esc_html_e('Emails with this template will not be sent.', 'paid-memberships-pro');?></p>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:56]:
```suggestion
					<p id="disable_description" class="description"><?php esc_html_e('Emails with this template will not be sent.', 'paid-memberships-pro'); ?></p>

```

File: adminpages/emailtemplates.php [hunk 6]
```diff
@@ -19,229 +19,244 @@
...
+				</td>
+			</tr>
+			<tr class="hide-while-loading">
+				<th scope="row" valign="top"><label for="email_template_subject"><?php esc_html_e('Subject', 'paid-memberships-pro');?></label></th>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:60]:
```suggestion
				<th scope="row" valign="top"><label for="email_template_subject"><?php esc_html_e( 'Subject', 'paid-memberships-pro' ); ?></label></th>

```

File: adminpages/emailtemplates.php [hunk 7]
```diff
@@ -19,229 +19,244 @@
...
+				<td>
+					<input id="email_template_subject" name="email_template_subject" type="text" size="100"/>
+				</td>
+			</tr>
+			<tr class="hide-while-loading">
+				<th scope="row" valign="top"><label for="email_template_body"><?php esc_html_e('Body', 'paid-memberships-pro');?></label></th>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:66]:
```suggestion
				<th scope="row" valign="top"><label for="email_template_body"><?php esc_html_e( 'Body', 'paid-memberships-pro' ); ?></label></th>

```

File: adminpages/emailtemplates.php [hunk 8]
```diff
@@ -19,229 +19,244 @@
...
+					</div>
+				</td>
+			</tr>
+			<tr class="hide-while-loading">
+				<th scope="row" valign="top"></th>
+				<td>
+					<?php esc_html_e( 'Send a test email to ', 'paid-memberships-pro' ); ?>
+					<input id="test_email_address" name="test_email_address" type="text"
+						value="<?php echo esc_attr( $current_user->user_email ); ?>"/>
+					<input id="send_test_email" class="button" name="send_test_email" value="<?php esc_attr_e('Save Template and Send Email', 'paid-memberships-pro');?>"
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:79]:
```suggestion
					<input id="send_test_email" class="button" name="send_test_email" value="<?php esc_attr_e( 'Save Template and Send Email', 'paid-memberships-pro' ); ?>"

```

File: adminpages/emailtemplates.php [hunk 9]
```diff
@@ -19,229 +19,244 @@
...
+						<?php esc_html_e( 'Your current membership will be used for any membership level data.', 'paid-memberships-pro' ); ?>
+					</p>
+				</td>
+			</tr>
+			<tr class="controls hide-while-loading">
+				<th scope="row" valign="top"></th>
+				<td>
+					<p class="submit">
+						<input id="submit_template_data" name="save_template" type="button" class="button-primary"
+							value="Save Template"/>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:92]:
```suggestion
							value="<?php esc_attr_e( 'Save Template', 'paid-memberships-pro' ); ?>"/>

```

File: adminpages/emailtemplates.php [hunk 10]
```diff
@@ -19,229 +19,244 @@
...
+						<input id="reset_template_data" name="reset_template" type="button" class="button"
+							value="Reset Template"/>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:94]:
```suggestion
							value="<?php esc_attr_e( 'Reset Template', 'paid-memberships-pro' ); ?>"/>

```

File: adminpages/emailtemplates.php [hunk 11]
```diff
@@ -19,229 +19,244 @@
...
-	<tr class="hide-while-loading">
-		<th scope="row" valign="top"></th>
-		<td>
-			<?php esc_html_e( 'Send a test email to ', 'paid-memberships-pro' ); ?>
-			<input id="test_email_address" name="test_email_address" type="text"
-				value="<?php echo esc_attr( $current_user->user_email ); ?>"/>
-			<input id="send_test_email" class="button" name="send_test_email" value="<?php esc_attr_e('Save Template and Send Email', 'paid-memberships-pro');?>"
-				type="button"/>
+		<div class="pmpro-email-templates-variable-reference">
+			<h1><?php esc_html_e('Variable Reference', 'paid-memberships-pro');?></h1>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:103]:
```suggestion
			<h1><?php esc_html_e( 'Variable Reference', 'paid-memberships-pro' ); ?></h1>

```

File: adminpages/emailtemplates.php [hunk 12]
```diff
@@ -19,229 +19,244 @@
...
+									<td><?php esc_html_e('User Email', 'paid-memberships-pro');?></td>
+								</tr>
+								<tr>
+									<td>!!login_link!!</td>
+									<td><?php esc_html_e('Login URL', 'paid-memberships-pro');?></td>
+								</tr>
+								<tr>
+									<td>!!levels_link!!</td>
+									<td><?php esc_html_e('Membership Levels Page URL', 'paid-memberships-pro');?></td>
+								</tr>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:163]:
```suggestion
								<?php
								$email_variables = [
									'!!name!!'                  => __( 'Display Name (Profile/Edit User > Display name publicly as)', 'paid-memberships-pro' ),
									'!!user_login!!'            => __( 'Username', 'paid-memberships-pro' ),
									'!!sitename!!'              => __( 'Site Title', 'paid-memberships-pro' ),
									'!!siteemail!!'             => __( 'Site Email Address (General Settings > Email OR Memberships > Settings > Email Settings)', 'paid-memberships-pro' ),
									'!!membership_id!!'         => __( 'Membership Level ID', 'paid-memberships-pro' ),
									'!!membership_level_name!!' => __( 'Membership Level Name', 'paid-memberships-pro' ),
									'!!membership_change!!'     => __( 'Membership Level Change', 'paid-memberships-pro' ),
									'!!membership_expiration!!' => __( 'Membership Level Expiration', 'paid-memberships-pro' ),
									'!!display_name!!'          => __( 'Display Name (Profile/Edit User > Display name publicly as)', 'paid-memberships-pro' ),
									'!!enddate!!'               => __( 'User Subscription End Date', 'paid-memberships-pro' ),
									'!!user_email!!'            => __( 'User Email', 'paid-memberships-pro' ),
									'!!login_link!!'            => __( 'Login URL', 'paid-memberships-pro' ),
									'!!levels_link!!'           => __( 'Membership Levels Page URL', 'paid-memberships-pro' ),
								];
								
								foreach ( $email_variables as $email_variable => $description ) {
									?>
										<tr>
											<td><?php echo esc_html( $email_variable ); ?></td>
											<td><?php echo esc_html( $description ); ?></td>
										</tr>
									<?php
								}
								?>
```

File: adminpages/emailtemplates.php [hunk 13]
```diff
@@ -19,229 +19,244 @@
...
+									<td><?php esc_html_e('Invoice Date', 'paid-memberships-pro');?></td>
+								</tr>
+								<tr>
+									<td>!!discount_code!!</td>
+									<td><?php esc_html_e('Discount Code Applied', 'paid-memberships-pro');?></td>
+								</tr>
+								<tr>
+									<td>!!invoice_link!!</td>
+									<td><?php esc_html_e('Invoice Page URL', 'paid-memberships-pro');?></td>
+								</tr>
```
Comment (sc0ttkclark) [inline adminpages/emailtemplates.php:248]:
```suggestion
								<?php
								$email_variables = [
									'!!billing_address!!' => __( 'Billing Info Complete Address', 'paid-memberships-pro' ),
									'!!billing_name!!'    => __( 'Billing Info Name', 'paid-memberships-pro' ),
									'!!billing_street!!'  => __( 'Billing Info Street Address', 'paid-memberships-pro' ),
									'!!billing_city!!'    => __( 'Billing Info City', 'paid-memberships-pro' ),
									'!!billing_state!!'   => __( 'Billing Info State', 'paid-memberships-pro' ),
									'!!billing_zip!!'     => __( 'Billing Info ZIP Code', 'paid-memberships-pro' ),
									'!!billing_country!!' => __( 'Billing Info Country', 'paid-memberships-pro' ),
									'!!billing_phone!!'   => __( 'Billing Info Phone #', 'paid-memberships-pro' ),
									'!!cardtype!!'        => __( 'Credit Card Type', 'paid-memberships-pro' ),
									'!!accountnumber!!'   => __( 'Credit Card Number (last 4 digits)', 'paid-memberships-pro' ),
									'!!expirationmonth!!' => __( 'Credit Card Expiration Month (mm format)', 'paid-memberships-pro' ),
									'!!expirationyear!!'  => __( 'Credit Card Expiration Year (yyyy format)', 'paid-memberships-pro' ),
									'!!membership_cost!!' => __( 'Membership Level Cost Text', 'paid-memberships-pro' ),
									'!!instructions!!'    => __( 'Payment Instructions (used in Checkout - Email Template)', 'paid-memberships-pro' ),
									'!!invoice_id!!'      => __( 'Invoice ID', 'paid-memberships-pro' ),
									'!!invoice_total!!'   => __( 'Invoice Total', 'paid-memberships-pro' ),
									'!!invoice_date!!'    => __( 'Invoice Date', 'paid-memberships-pro' ),
									'!!discount_code!!'   => __( 'Discount Code Applied', 'paid-memberships-pro' ),
									'!!invoice_link!!'    => __( 'Invoice Page URL', 'paid-memberships-pro' ),
								];

								foreach ( $email_variables as $email_variable => $description ) {
									?>
										<tr>
											<td><?php echo esc_html( $email_variable ); ?></td>
											<td><?php echo esc_html( $description ); ?></td>
										</tr>
									<?php
								}
								?>
```

File: js/pmpro-admin.js [hunk 14]
```diff
@@ -373,16 +373,16 @@ jQuery(document).ready(function($) {
...
             if(response['result'] == false) {
-                $("#message").addClass("error");
+                $(".status_message_wrapper").addClass("error");
                 $(".status_message").show().text("There was an error updating your template settings.");
             }
             else {
                 if(response['status'] == 'true') {
-                    $("#message").addClass("updated");
+                    $(".status_message_wrapper").addClass("updated");
                     $(".status_message").show().text("Template Disabled");
```
Comment (sc0ttkclark) [inline js/pmpro-admin.js:382]: We're going to have to sort out these hardcoded english strings in the future

File: includes/deprecated.php [hunk 15]
```diff
@@ -96,7 +96,7 @@ function pmpro_check_for_deprecated_add_ons() {
 		}
 		?>
 		<div class="notice notice-warning">
-        <p><?php _e( sprintf( 'Some of the PMPro Add Ons on your site have been merged into the core PMPro plugin or are otherwise not needed. <br />The following plugins <strong>should be deleted</strong> from your site: <strong>%s</strong>.', implode( ', ', $deprecated_active ) ), 'paid-memberships-pro' ); ?></p>
+        <p><?php _e( sprintf( 'Some Add Ons are now merged into the Paid Memberships Pro core plugin. The features of the following plugins are now included in PMPro by default. You should <strong>delete these unnecessary plugins</strong> from your site: <em><strong>%s</strong></em>.', implode( ', ', $deprecated_active ) ), 'paid-memberships-pro' ); ?></p>
```
Comment (sc0ttkclark) [inline includes/deprecated.php:99]:
```suggestion
		<p>
			<?php
				// translators: %s: The list of deprecated plugins that are active.
				printf(
					__( 'Some Add Ons are now merged into the Paid Memberships Pro core plugin. The features of the following plugins are now included in PMPro by default. You should <strong>delete these unnecessary plugins</strong> from your site: <em><strong>%s</strong></em>.', 'paid-memberships-pro' ),
					implode( ', ', $deprecated_active )
				);
			?>
		</p>
```

**Review by sc0ttkclark — COMMENTED** [review]
A few changes / improvements, we can always abstract the variable reference further but at least with my suggestion it will be on that path

**Pull request merged**